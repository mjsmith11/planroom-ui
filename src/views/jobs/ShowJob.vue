<template>
<div>
  <logo-header v-if="!$store.getters.isContractorUser"></logo-header>
  <div class="container">
    <div class="col-xs-12 col-sm-10 col-md-8 offset-md-2 offset-sm-1 column">
        <h1>{{ job.name }}</h1>
        <h6 v-if="!$store.getters.isContractorUser" class="expiration">You can access this job unitl {{ $store.getters.exp | unixDatetime | datetime }} </h6>
        <router-link v-else :to="'/jobs/' + this.$route.params.id + '/invite'" tag="button" class="btn btn-outline-primary btn-sm btn-invite">Invite Subcontractors</router-link>
        <div class="row">
          <div class="col-lg-4 col-md-6 col-sm-12 column">
            <div class="card">
              <h5 class="card-header">Bid</h5>
              <div class="card-body">
                <div class="card-text">
                  <h6><strong>Date:</strong></h6>
                  <div>{{ job.bidDate | date }}</div>
                  <br>
                  <h6><strong>Email:</strong></h6>
                  <div> <a :href="'mailto:'+job.bidEmail">{{ job.bidEmail }}</a> </div>
                  <br>
                  <h6><strong>Due To Benchmark:</strong></h6>
                  <div>{{ job.subcontractorBidsDue | datetime }}</div>
                  <div>{{ job.subcontractorBidsDue | datetimeCentral }}</div>
                </div>
              </div>
            </div>
          </div>
          <div class="col-lg-4 col-md-6 col-sm-12 column">
            <div class="card">
              <h5 class="card-header">Prebid</h5>
              <div class="card-body">
                <div class="card-text">
                  <h6><strong>Date &amp; Time:</strong></h6>
                  <div>{{ job.prebidDateTime | datetime }}</div>
                  <div>{{ job.prebidDateTime | datetimeCentral }}</div>
                  <br>
                  <h6><strong>Address</strong></h6>
                  <div>{{ job.prebidAddress }}</div>
                </div>
              </div>
            </div>
          </div>
          <div class="col-lg-4 col-md-6 col-sm-12 column">
            <div class="card">
              <h5 class="card-header">Details</h5>
                <div class="card-body">
                  <div class="card-text">
                    <h6><strong>Taxable</strong></h6>
                    <div>{{ job.taxible | TFtoYN}}</div>
                    <br>
                    <h6><strong>Subcontractor Bonding</strong></h6>
                    <div>{{ job.bonding | TFtoYN }}</div>
                  </div>
                </div>
            </div>
          </div>
        </div>
        <div class="row">
          <plan-lister :job-id = job.id></plan-lister>
        </div>
        <div class="row" v-if="$store.getters.isContractorUser">
          <uploader :jobId = job.id></uploader>
        </div>
    </div>
    <contact-info v-if="!$store.getters.isContractorUser"></contact-info>
  </div>
</div>
</template>
<script>
import { datetimeFilters } from '../../mixins/datetimeFilters'
import uploader from '../../components/plans/Uploader'
import lister from '../../components/plans/Lister'
import axios from 'axios'
import { EventBus } from '../../event-bus.js'
import LogoHeader from '../../components/LogoHeader'
import ContactInfo from '../../components/ContactInfo'
export default {
  data () {
    return {
      job: {}
    }
  },
  components: {
    uploader: uploader,
    planLister: lister,
    logoHeader: LogoHeader,
    contactInfo: ContactInfo
  },
  mixins: [datetimeFilters],
  filters: {
    TFtoYN (value) {
      if (value !== 1 && value !== 0) {
        return ''
      }
      return value ? 'Yes' : 'No'
    }
  },
  created () {
    axios.get('/jobs/' + this.$route.params.id)
      .then(res => {
        this.job = res.data
        EventBus.$emit('job-read')
      })
      // eslint-disable-next-line
      .catch(err => {
        // console.log(err)
      })
  }
}
</script>
<style scoped>
  div.column {
    padding: 3px;
  }
  .card {
    height: 100%;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.19);
  }
  h1 {
    font-style: italic;
    font-weight: 600;
  }
  .expiration {
    font-style: italic;
    font-size: 0.85em;
    color: #b0b0b0;
  }
  .btn-invite{
    margin-bottom: 15px;
  }
</style>
