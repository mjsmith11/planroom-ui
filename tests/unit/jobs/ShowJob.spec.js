import { mount, createLocalVue } from '@vue/test-utils'
import ShowJob from '@/views/jobs/ShowJob'
import mockAxios from 'jest-mock-axios'
import { EventBus } from '@/event-bus.js'
import Vuex from 'vuex'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('Show Job', () => {
  let cmp
  let getters = {
    token: () => 'eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJleHAiOjEyMzQ1Njc3ODc3fQ.6xV-z88Nvmag8i4jVwmOZjX3MhCYAgb3rqttN4ROix3EbtHLwYIG3utNVaCpCN2cS7QFAJM3CPnfiS5_s9luiA'
  }
  let store = new Vuex.Store({
    getters
  })
  const $route = {
    params: {
      id: 12
    }
  }

  beforeEach(() => {
    cmp = mount(ShowJob, {
      mocks: {
        $route
      },
      store,
      localVue
    })
  })

  afterEach(() => {
    // cleaning up the mess left behind the previous test
    mockAxios.reset()
  })
  describe('Bool filter', () => {
    it('detects invalid input', () => {
      expect(cmp.vm.$options.filters.TFtoYN(2)).toBe('')
    })

    it('returns Yes', () => {
      expect(cmp.vm.$options.filters.TFtoYN(1)).toBe('Yes')
    })

    it('returns No', () => {
      expect(cmp.vm.$options.filters.TFtoYN(0)).toBe('No')
    })
  })

  it('has expected html structure', () => {
    const response = {
      data: {
        'bidDate': '2017-03-23',
        'bidEmail': 'abc@xyz.com',
        'bonding': 0,
        'name': 'abc',
        'prebidAddress': '123 Main St.',
        'prebidDateTime': '2018-07-01 18:30:00',
        'subcontractorBidsDue': '2017-06-01 08:30:00',
        'taxible': 1,
        'id': 25
      }
    }
    mockAxios.mockResponse(response)
    expect(cmp.html()).toMatchSnapshot()
  })

  it('processes api response', () => {
    let evtHandler = jest.fn()
    EventBus.$on('job-read', evtHandler)
    expect(mockAxios.get).toHaveBeenCalledWith('/jobs/12')
    expect(mockAxios.get).toHaveBeenCalledTimes(1)

    const response = {
      data: {
        'bidDate': '2017-12-23',
        'bidEmail': 'abcdef@xyz.com',
        'bonding': false,
        'name': 'My Second Test Job',
        'prebidAddress': '234 Main St.',
        'prebidDateTime': '2018-07-01 18:30:00',
        'subcontractorBidsDue': '2017-06-01 08:30:00',
        'taxible': true,
        'id': 20
      }
    }
    mockAxios.mockResponse(response)
    expect(cmp.vm.job).toBe(response.data)
    expect(evtHandler).toHaveBeenCalledTimes(1)
  })

  // TODO add tests for error handling when it's implemented
})
