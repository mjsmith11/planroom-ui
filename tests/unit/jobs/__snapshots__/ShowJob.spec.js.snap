// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Job to Contractor has expected html structure 1`] = `
<div>
  <!---->
  <div class="container">
    <div class="col-xs-12 col-sm-10 col-md-8 offset-md-2 offset-sm-1 column">
      <h1>abc</h1>
      <!---->
      <div class="row">
        <div class="col-lg-4 col-md-6 col-sm-12 column">
          <div class="card">
            <h5 class="card-header">Bid</h5>
            <div class="card-body">
              <div class="card-text">
                <h6><strong>Date:</strong></h6>
                <div>03/23/2017</div>
                <br>
                <h6><strong>Email:</strong></h6>
                <div>abc@xyz.com</div>
                <br>
                <h6><strong>Due To Benchmark:</strong></h6>
                <div>06/01/2017 08:30 AM EDT</div>
                <div>06/01/2017 07:30 AM CDT</div>
              </div>
            </div>
          </div>
        </div>
        <div class="col-lg-4 col-md-6 col-sm-12 column">
          <div class="card">
            <h5 class="card-header">Prebid</h5>
            <div class="card-body">
              <div class="card-text">
                <h6><strong>Date &amp; Time:</strong></h6>
                <div>07/01/2018 06:30 PM EDT</div>
                <div>07/01/2018 05:30 PM CDT</div>
                <br>
                <h6><strong>Address</strong></h6>
                <div>123 Main St.</div>
              </div>
            </div>
          </div>
        </div>
        <div class="col-lg-4 col-md-6 col-sm-12 column">
          <div class="card">
            <h5 class="card-header">Details</h5>
            <div class="card-body">
              <div class="card-text">
                <h6><strong>Taxable</strong></h6>
                <div>Yes</div>
                <br>
                <h6><strong>Subcontractor Bonding</strong></h6>
                <div>No</div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="row">
        <div class="lister">
          <h4>Bidding Documents</h4>
          <ul class=""></ul>
        </div>
      </div>
      <div class="row">
        <div class="uploader">
          <h4>Add Bidding Documents</h4>
          <!---->
          <div id="" class="vue-dropzone dropzone dz-clickable">
            <!---->
            <div class="dz-default dz-message"><span>Drop files here to upload</span></div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
</div>
`;

exports[`Show Job to Contractor shows job for subcontractor 1`] = `
<div>
  <div class="jumbotron">
    <div class="row">
      <h4 class="offset-md-1 col-md-10">Welcome To The</h4>
      <div class="offset-md-1 col-md-10">
        <img src="../assets/logo1.png" alt="Benchmark logo" class="src">
      </div>
      <div class="offset-md-1 col-md-10 text">
        <h1>Planroom</h1>
      </div>
    </div>
  </div>
  <div class="container">
    <div class="col-xs-12 col-sm-10 col-md-8 offset-md-2 offset-sm-1 column">
      <h1></h1>
      <h6 class="expiration">You can access this job unitl 11/7/2018 10:50 PM EST </h6>
      <div class="row">
        <div class="col-lg-4 col-md-6 col-sm-12 column">
          <div class="card">
            <h5 class="card-header">Bid</h5>
            <div class="card-body">
              <div class="card-text">
                <h6><strong>Date:</strong></h6>
                <div></div>
                <br>
                <h6><strong>Email:</strong></h6>
                <div></div>
                <br>
                <h6><strong>Due To Benchmark:</strong></h6>
                <div></div>
                <div></div>
              </div>
            </div>
          </div>
        </div>
        <div class="col-lg-4 col-md-6 col-sm-12 column">
          <div class="card">
            <h5 class="card-header">Prebid</h5>
            <div class="card-body">
              <div class="card-text">
                <h6><strong>Date &amp; Time:</strong></h6>
                <div></div>
                <div></div>
                <br>
                <h6><strong>Address</strong></h6>
                <div></div>
              </div>
            </div>
          </div>
        </div>
        <div class="col-lg-4 col-md-6 col-sm-12 column">
          <div class="card">
            <h5 class="card-header">Details</h5>
            <div class="card-body">
              <div class="card-text">
                <h6><strong>Taxable</strong></h6>
                <div></div>
                <br>
                <h6><strong>Subcontractor Bonding</strong></h6>
                <div></div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="row">
        <div class="lister">
          <h4>Bidding Documents</h4>
          <ul class=""></ul>
        </div>
      </div>
      <!---->
    </div>
    <div class="row">
      <div class="col-lg-2 offset-lg-4 col-md-12">
        <div class="info">
          <font-awesome-icon-stub icon="map-marker-alt"></font-awesome-icon-stub>&nbsp;&nbsp;318 E. Monroe St.</div>
        <div class="info">Kokomo, IN 46901</div>
      </div>
      <div class="col-lg-2 col-md-12">
        <div class="info">
          <font-awesome-icon-stub icon="phone"></font-awesome-icon-stub>&nbsp;&nbsp;(765) 452-9500</div>
        <div class="info">
          <font-awesome-icon-stub icon="fax"></font-awesome-icon-stub>&nbsp;&nbsp;(765) 452-9505</div>
      </div>
    </div>
  </div>
</div>
`;
